ecl/eclagent/eclagent.cpp

```diff
@@ -1950,6 +1951,12 @@ void EclAgent::doProcess()
             break;
         }
 
+        if (w->getState() == WUStateCompleted)
```

Comment:
We should have a #option that can be used to disable this code - just in case there is a problem with it that causes eclagent to crash.